src/thermo/PengRobinsonMFTP.cpp

```diff
@@ -0,0 +1,1300 @@
+                        * (dalphadT_vec_Curr_[j] * alpha_vec_Curr_[i] + dalphadT_vec_Curr_[i] * alpha_vec_Curr_[j]);
+        }
+    }
+    return daAlphadT;
+}
+
+double PengRobinsonMFTP::d2aAlpha_dT2() const
+{
+    double daAlphadT = 0.0, temp, fac1, fac2, alphaij, alphai, alphaj, d2aAlphadT2 = 0.0, num;
+    double k;
+    double sqt_Tr = sqrt(temperature() / critTemperature()); //we need species critical temperature
+    double coeff1 = 1 / (critTemperature()*critTemperature()*sqt_Tr);
+    double coeff2 = sqt_Tr - 1;
+    for (size_t i = 0; i < m_kk; i++) {
+        //  Calculate first and second derivatives of alpha for individual species
+        size_t counter = i + m_kk * i;
+        k = kappa_vec_[i];
+        dalphadT_vec_Curr_[i] = coeff1 *(k* k*coeff2 - k);
+        d2alphadT2_[i] = (k*k + k) * coeff1 / (2 * sqt_Tr*sqt_Tr);
+    }
+
+    //Calculate mixture derivative
+    for (size_t i = 0; i < m_kk; i++) {
+        size_t counter1 = i + m_kk * i;
+        alphai = alpha_vec_Curr_[i];
+        for (size_t j = 0; j < m_kk; j++) {
+            size_t counter2 = j + m_kk * j;
+            alphaj = alpha_vec_Curr_[j];
+            alphaij = alphai * alphaj;
+            temp = 0.5 * sqrt((a_vec_Curr_[counter1] * a_vec_Curr_[counter2]) / (alphaij));
+            num = (dalphadT_vec_Curr_[j] * alphai + dalphadT_vec_Curr_[i] * alphaj);
+            fac1 = -(0.5 / alphaij)*num*num;
+            fac2 = alphaj * d2alphadT2_[counter1] + alphai *d2alphadT2_[counter2] + 2 * dalphadT_vec_Curr_[i] * dalphadT_vec_Curr_[j];
+            d2aAlphadT2 += moleFractions_[i] * moleFractions_[j] * temp *(fac1 + fac2);
+        }
+    }
+    return d2aAlphadT2;
+}
+
+void PengRobinsonMFTP::calcCriticalConditions(double a, double b,
+        double& pc, double& tc, double& vc) const
+{
+    if (b <= 0.0) {
+        tc = 1000000.;
+        pc = 1.0E13;
+        vc = omega_vc * GasConstant * tc / pc;
+        return;
+    }
+    if (a <= 0.0) {
+        tc = 0.0;
+        pc = 0.0;
+        vc = 2.0 * b;
+        return;
+    }
+    tc = a * omega_b / (b * omega_a * GasConstant);
+    pc = omega_b * GasConstant * tc / b;
+    vc = omega_vc * GasConstant * tc / pc;
+}
+
+int PengRobinsonMFTP::NicholsSolve(double TKelvin, double pres, double a, double b, double aAlpha,
+                                   double Vroot[3]) const
+{
+    double tmp;
+    fill_n(Vroot, 3, 0.0);
+    if (TKelvin <= 0.0) {
+        throw CanteraError("PengRobinsonMFTP::NicholsSolve()", "negative temperature T = {}", TKelvin);
+    }
+
+    // Derive the coefficients of the cubic polynomial (in terms of molar volume v) to solve.
+    double bsqr = b * b;
+    double RT_p = GasConstant * TKelvin / pres;
+    double aAlpha_p = aAlpha / pres;
+    double an = 1.0;
+    double bn = (b - RT_p);
+    double cn = -(2 * RT_p * b - aAlpha_p + 3 * bsqr);
+    double dn = (bsqr * RT_p + bsqr * b - aAlpha_p * b);
+
+    double tc = a * omega_b / (b * omega_a * GasConstant);
```

Comment:
@gkogekar in Redlich Kwong, this value is raised to the 2/3 power.  Can you comment on the discrepancy, here?  Can you verify whether this and/or Redlich Kwong are correct?

Reply:
Okay, just went and reviewed the EoS, and can confirm that the implementation is at least correct for `Peng-Robinson`.

Reply:
@decaluwe The implementation for the Redlich-Kwong is also correct.  I have verified that.